gossip/emitter/prev_action_files.go

```diff
@@ -82,24 +71,35 @@ func (em *Emitter) writeLastEmittedEpochVote(e idx.Epoch) {
 	if em.emittedEvFile == nil {
 		return
 	}
-	_, err := em.emittedEvFile.WriteAt(e.Bytes(), 0)
-	if err != nil {
+	if _, err := em.emittedEvFile.WriteAt(e.Bytes(), 0); err != nil {
 		log.Crit("Failed to write BVs file", "file", em.config.PrevEpochVoteFile.Path, "err", err)
 	}
 }
 
 func (em *Emitter) readLastEpochVote() *idx.Epoch {
-	if em.emittedEvFile == nil {
+	buf := readFile(em.emittedEvFile, 4, "Failed to read EV file", em.config.PrevEpochVoteFile.Path, )
+	if buf == nil {
 		return nil
 	}
-	buf := make([]byte, 4)
-	_, err := em.emittedEvFile.ReadAt(buf, 0)
-	if err != nil {
+	
+	v := idx.BytesToEpoch(buf)
+	return &v
+}
+
+// should we put logger on *Emitter?
+func readFile(file *os.File, bufLen uint8, errMsg, path string) []byte {
+	if file == nil {
+		return nil
+	}
+
+	buf := make([]byte, bufLen)
+	if _, err := file.ReadAt(buf, 0); err != nil {
```

Comment:
Technically we had/have a bug here, because ReadAt is not obliged to read the entire buffer. Practically it works only because buffer is very small

Reply:
We may use `Seek` and then call `ioread.ReadAll` to read entire buffer. Or, we can wrap it with io.NewSectionReader (so that we wouldn't need to call `Seek`) and then call `ioread.ReadAll`

Reply:
Oh sorry, I'm wrong:
```
// When ReadAt returns n < len(p), it returns a non-nil error
// explaining why more bytes were not returned. In this respect,
// ReadAt is stricter than Read.
```